extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/JdbcMetaStoreManagerFactory.java

```diff
@@ -45,14 +46,15 @@ public class JdbcMetaStoreManagerFactory extends LocalPolarisMetaStoreManagerFac
 
   // TODO: Pending discussion of if we should have one Database per realm or 1 schema per realm
   // or realm should be a primary key on all the tables.
-  @Inject DataSource dataSource;
+  @Inject Instance<DataSource> dataSource;
```

Comment:
Why do we need `Instance` here?

In any case, custom CDI-related resolvers should probably be in the Quarkus-specific modules.

Reply:
This is Quarkus failing the Reg test saying the bean is not active as we don't supply the jdbc.url and hence promting me to use Instance here to avoid this failure.

Reply:
Noted. In this case, I'd say we have to refactor this and have a "producer" in quarkus-specific modules to allow clean injection int the jdbc module (i.e. quarkus-specific code can use `Instance`, but believe is it best avoided here).

Reply:
Let's revisit this as discussed offline. If the JDBC MetaStore factory is not selected by the `type` config, it should not affect CDI injections (should not fail startup). If JDBC is selected, then the DataSource should be configured. 

In any case, if we have to use `Instance`, it should not be handled here, but in Quarkus-specific code (producers).